chapters/en/slides/module5/module5_03.Rmd

```diff
@@ -0,0 +1,316 @@
+
+---
+
+## We can zoom in to highlight a region of an axes
+
+```{python}
+title_text = 'Wikipedia donations peak during Christmas'
+alt.Chart(donations[donations['date'].between("2020", "2020-07")], title=title_text).mark_circle(clip=True).encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day', scale=alt.Scale(domain=[0, 100_000])))
+```
+
+Notes:
+It is not always practical to filter the data,
+so it is good to know how to change the axis range directly in Altair.
+For this,
+we could modify the domain of the axis' scale.
+
+When we used `alt.Axis` previously,
+the changes we made were cosmetic,
+such as tick numbers,
+label formatting etc.
+
+Now that were are modifying the range of the actual displayed values,
+we need to use `alt.Scale` instead.
+It also works for colors and sizes in a legend,
+just like `alt.Axis`.
+
+Finally we need to set `clip=True` in the axis,
+to be explicit about that we are excluding data.
+Without it,
+we would still see the circles extending beyond the range of the chart.
+
+This ensures that we are intentional about our data exclusion,
+and it is something we should think twice about because it can skew the perception of the data.
+For example,
+as we saw in the previous slide deck,
+we should not do in to a bar chart that shows counts or proportions.
+
+After zooming in on the y-axis,
+we can more clearly see the variation in the lower value data points.
+
+---
+
+## Axes can be anchored to the data range instead of to zero without explicitly setting the domain
+
+```{python}
+alt.Chart(donations[donations['sum'] > 2.5e6]).mark_circle().encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day'))
+```
+
+Notes:
+You might have noticed from previous modules that Altair often includes 0 in the axis,
+even if our data does not have any values close to 0.
+The reason for this is to enforce good habits 
+and not create for example bar plot that are cut off
+which makes difference appear much bigger than they actually are.
+
+However,
+there are also good reasons for setting the range based on the extent of the data,
+e.g. do be able to show the variation fully.
+
+In this slide we can see that showing only the days with high total donations
+only uses about half of the vertical space available in the chart.
+
+---
+
+## Axis can be anchored to the data range instead of to zero without explicitly setting the domain
+
+```{python}
+alt.Chart(donations[donations['sum'] > 2.5e6]).mark_circle().encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day', scale=alt.Scale(zero=False)))
+```
+
+Notes:
+If we wanted to use the full range,
+we could set the domain explicitly like before,
+but the shortcut `zero=False` allows us to adjust the axis
+to the min and max values of the data with less typing.
```

Comment:
Here I think we need to caution them that we often shouldn't do this - as it makes differences seem bigger than they actually are...